airflow/www/views.py

```diff
@@ -2385,6 +2385,18 @@ class DagRunModelView(ModelViewOnly):
         start_date=datetime_f,
         dag_id=dag_link)
 
+    @expose('/edit/', methods=('GET', 'POST'))
+    def edit_view(self):
+        return_value = super(DagRunModelView, self).edit_view()
+        if request and request.method == 'POST' and request.form.get('state') in [State.SUCCESS, State.FAILED]:
```

Comment:
@dangermike this won't do this update for `State.RUNNING`. I want to remove that condition since the behavior changes for different states which is weird. Let me know if you think this is a bad idea

Reply:
I honestly don't know. Sounds reasonable, but this whole exercise has shown that _reasonable_ and _correct_ don't always overlap.